packages/server/modules/workspaces/repositories/workspaces.ts

```diff
@@ -35,21 +38,57 @@ export const upsertWorkspaceFactory =
       .merge(['description', 'logoUrl', 'name', 'updatedAt'])
   }
 
-export const getWorkspaceRoleFactory =
-  ({ db }: { db: Knex }): GetWorkspaceRole =>
+export const getWorkspaceRolesFactory =
+  ({ db }: { db: Knex }): GetWorkspaceRoles =>
+  async ({ workspaceId }) => {
+    return await tables.workspacesAcl(db).select('*').where({ workspaceId })
+  }
+
+export const getWorkspaceRoleForUserFactory =
+  ({ db }: { db: Knex }): GetWorkspaceRoleForUser =>
+  async ({ userId, workspaceId }) => {
+    return (
+      (await tables
+        .workspacesAcl(db)
+        .select('*')
+        .where({ userId, workspaceId })
+        .first()) ?? null
+    )
+  }
+
+export const getWorkspaceRolesForUserFactory =
+  ({ db }: { db: Knex }): GetWorkspaceRolesForUser =>
+  async ({ userId }, options) => {
+    const workspaceIdFilter = options?.workspaceIdFilter ?? []
+
+    const query = tables.workspacesAcl(db).select('*').where({ userId })
+
+    if (workspaceIdFilter.length > 0) {
+      query.whereIn('workspaceId', workspaceIdFilter)
+    }
+
+    return await query
+  }
+
+export const deleteWorkspaceRoleFactory =
+  ({ db }: { db: Knex }): DeleteWorkspaceRole =>
   async ({ userId, workspaceId }) => {
-    const acl = await tables
+    const deletedRoles = await tables
       .workspacesAcl(db)
-      .select('*')
-      .where({ userId, workspaceId })
-      .first()
+      .where({ workspaceId, userId })
+      .delete('*')
+
+    if (deletedRoles.length === 0) {
+      return null
+    }
 
-    return acl || null
+    return deletedRoles[0]
```

Comment:
just make a note, that this is safe always, cause `workspaceId` and `userId` are composing a PK on the table, that ensures uniqueness. So there is always 0 or 1 entry returned here.